### PR TITLE
CNN section 6.1 - 6.3

### PR DESCRIPTION
**Pull request by aksrajvanshi**
Addition: 

Adding on CNN sections from 6.1 to 6.3

### CODE REVIEW

File: jupyter/d2l-java/chapter_convolutional-neural-networks/conv-layer.ipynb [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+    "TrainingConfig config = new DefaultTrainingConfig(Loss.l2Loss()).optOptimizer(sgd);\n",
+    "Model model = Model.newInstance(\"conv2D\");\n",
+    "model.setBlock(block);\n",
+    "Trainer trainer = model.newTrainer(config);\n",
+    "trainer.initialize(X.getShape());\n",
+    "\n",
+    "NDArray lossVal = null;\n",
+    "for(int i=0; i < 10; i++){\n",
+    "    \n",
+    "     try (GradientCollector gc = Engine.getInstance().newGradientCollector()) {\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-neural-networks/conv-layer.ipynb:409]: This example seems to be half using Trainer and half from scratch. I think either way would be fine, but be consistent with the choice

File: jupyter/d2l-java/chapter_convolutional-neural-networks/conv-layer.ipynb [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    "\n",
+    "Loss l2Loss = Loss.l2Loss();\n",
+    "\n",
+    "LearningRateTracker lrt = LearningRateTracker.fixedLearningRate(0.4f);\n",
+    "Optimizer sgd = Optimizer.sgd().setLearningRateTracker(lrt).build();\n",
+    "\n",
+    "TrainingConfig config = new DefaultTrainingConfig(Loss.l2Loss()).optOptimizer(sgd);\n",
+    "Model model = Model.newInstance(\"conv2D\");\n",
+    "model.setBlock(block);\n",
+    "Trainer trainer = model.newTrainer(config);\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-neural-networks/conv-layer.ipynb:403]: Can you add a newline before this to split these lines up a bit?

File: jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "NDArray X = manager.randomUniform(0f, 1.0f, new Shape(8, 8));\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb:174]: You can just specify the shape (1, 1, 8, 8) here instead of reshaping it on the next line

File: jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    "X = X.reshape(1, 1, 8, 8);\n",
+    "\n",
+    "Block block = Conv2D.builder()\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb:177]: What do you think about creating the variables in a separate code block between importing and this one. Then, this code block can look equivalent to the other ones to help emphasize what you are doing

File: jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+   },
+   "outputs": [],
+   "source": [
+    "block = Conv2D.builder()\n",
+    "                .setKernel(new Shape(5, 3))\n",
+    "                .optPad(new Shape(2, 1))\n",
+    "                .optBias(false)\n",
+    "                .setNumFilters(1)\n",
+    "                .build();\n",
+    "model.setBlock(block);\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb:222]: Add newline after block here too
